src/api/spec/models/badge_spec.rb

```diff
@@ -0,0 +1,77 @@
+require 'rails_helper'
+
+RSpec.describe Badge do
+  let(:user) { create(:confirmed_user, :with_home, login: 'tom') }
+  let(:source_project) { user.home_project }
+  let(:source_package) { create(:package, name: 'my_package', project: source_project) }
+  let(:request) do
+    stub_request(:get, "#{CONFIG['source_url']}/build/#{user.home_project}/_result?locallink=1&multibuild=1&lastbuild=1" \
+                       "&package=#{source_package}&view=status").and_return(body: body)
+  end
+  let(:results) { request && source_package.buildresult(source_project, false, true).results[source_package.name] }
+  let(:badge) { Badge.new(type, results) }
+
+  describe '#new' do
+    RSpec.shared_examples 'tests for badge xml' do
+      describe 'empty restultlist' do
```

Comment:
```suggestion
      describe 'empty resultlist' do
```
Typo. It happens a couple of times more in this same file.